src/nvmet.c

```diff
@@ -739,59 +1055,94 @@ int nvmet_view_ports(bool json_flag)
 		ports = ports->next;
 		free(tmp_ports);
 	}
-
+	free_nvmet_linked_lists(ports, NULL);
+	ports = NULL;
 	return SUCCESS;
 }
 
-int nvmet_enable_port(unsigned char *interface, int port, int protocol)
+/**
+ * It creates a new port.
+ *
+ * @param interface The interface to use for the port.
+ * @param port The port number to create.
+ * @param protocol 0 = TCP, 1 = RDMA
+ * @param return_message This is a pointer to a buffer that will be filled with the return message.
+ *
+ * @return The return value is the return code of the function.
+ */
+int nvmet_enable_port(char *interface, int port, int protocol, char *return_message)
 {
-        int rc = INVALID_VALUE;
+	int rc = INVALID_VALUE;
 	FILE *fp;
-        unsigned char path[NAMELEN] = {0x0}, ip[0xF] = {0x0}, proto[0x5] = {0x0};
+	char path[NAMELEN] = {0x0}, ip[0xF] = {0x0}, proto[0x5] = {0x0};
 	mode_t mode = (S_IRUSR | S_IWUSR | S_IXUSR | S_IRGRP | S_IXGRP | S_IROTH | S_IXOTH);
 	struct NVMET_PORTS *ports, *tmp_ports;
-        
-	ports = (struct NVMET_PORTS *)nvmet_scan_all_ports();
+	char *msg;
+	char error_message[NAMELEN] = {0};
+
+	ports = nvmet_scan_ports(error_message);
+	if ((ports == NULL) && (strlen(error_message) != 0)) {
+//		print_message(INVALID_VALUE, error_message, json_flag);
+		msg = "%s";
+		print_error(msg, return_message, error_message);
+		return INVALID_VALUE;
+	}
 
 	sprintf(path, "%s/%d", SYS_NVMET_PORTS, port);
 	if (access(path, F_OK) == SUCCESS) {
-		printf("Error. NVMe Target Port %d already exists.\n", port);
+		msg = "Error. NVMe Target Port %d already exists.";
+		print_error(msg, return_message, port);
 		return rc;
 	}
 
-	sprintf(ip, "%s", nvmet_interface_ip_get(interface));
+	char *ipaddr = nvmet_interface_ip_get(interface, return_message);
+	if (ipaddr == NULL) {
+//		msg = "Error. IP address %s is invalid.";
```

Comment:
Why do we not want to return a string message here?

Reply:
Fixed!